ros2doctor/ros2doctor/api/platform.py

```diff
@@ -14,87 +14,102 @@
 
 import os
 import platform
-import sys
+import warnings
 
-from ros2doctor.api.format import print_term
+from ros2doctor.api import DoctorCheck
+from ros2doctor.api import DoctorReport
+from ros2doctor.api import Report
+from ros2doctor.api.format import warning_format
 
 import rosdistro
 
 
-def print_platform_info():
-    """Print out platform information."""
-    platform_name = platform.system()
-    # platform info
-    print('PLATFORM INFORMATION')
-    print_term('system', platform_name)
-    print_term('platform info', platform.platform())
-    if platform_name == 'Darwin':
-        print_term('mac OS version', platform.mac_ver())
-    print_term('release', platform.release())
-    print_term('processor', platform.processor())
-
-    # python info
-    print('PYTHON INFORMATION')
-    print_term('version', platform.python_version())
-    print_term('compiler', platform.python_compiler())
-    print_term('build', platform.python_build())
-    print('\n')
-
-
-def check_platform_helper():
-    """Check ROS_DISTRO related environment variables and distribution name."""
+warnings.formatwarning = warning_format
```

Comment:
Recommend using a context manager to only override this while checks and reports are being run. This will make it more robust to something else being imported that also customizes `formatwarning`.

Something like

```python3

import warnings

def warning_format(msg, cat, filename, linenum, file=None, line=None):
    return 'testtesttest %s: %s: %s: %s\n' % (filename, linenum, cat.__name__, msg)


class CustomWarningFormat:

    def __enter__(self):
        self._old_format = warnings.formatwarning
        warnings.formatwarning = warning_format

    def __exit__(self, t, v, tb):
        warnings.formatwarning = self._old_format


# Run checks/reports inside `with` statement to limit who the warning is customized for

warnings.warn('old format')

with CustomWarningFormat():
    warnings.warn('custom format')

warnings.warn('old format again')
```

Results in 

```
$ python3 warn.py 
warn.py:18: UserWarning: old format
  warnings.warn('old format')
testtesttest warn.py: 21: UserWarning: custom format
warn.py:23: UserWarning: old format again
  warnings.warn('old format again')
```